server/py/framework/db/sqldb/db.py

```diff
@@ -7149,3 +7152,103 @@ def _paginate_query(
             query = query.limit(limit)
 
         return query
+
+    @staticmethod
+    def _find_artifact_latest_tag(session, object_record):
+        """Find the 'latest' tag for an artifact object."""
+        return (
+            session.query(ArtifactV2.Tag)
+            .filter(
+                ArtifactV2.Tag.obj_id == object_record.id,
+                ArtifactV2.Tag.name == mlrun.common.constants.RESERVED_TAG_NAME_LATEST,
+                ArtifactV2.Tag.project == object_record.project,
+            )
+            .one_or_none()
+        )
+
+    @staticmethod
+    def _find_other_latest_in_other_iterations(session, object_record):
```

Comment:
2 `other` is confusing
```suggestion
    def _find_latest_tag_in_other_iterations(session, object_record):
```

Reply:
@TomerShor I have changed the function name to `_has_latest_tag_in_different_iterations` because it now returns True or False to indicate whether the tag exists